pages/create.tsx

```diff
@@ -54,6 +62,71 @@ const Home: NextPage = () => {
     }
   }
 
+  interface NengajoTokenMetadata {
+    name: string
+    image: string
+    description?: string | null | undefined
+    animation_url?: string | null | undefined
+    external_url?: string | null | undefined
+    attributes: TokenAttribute[]
+  }
+  interface TokenAttribute {
+    trait_type: string
+    value: number | string
+    display_type?: string | null | undefined
+    max_value?: number | null | undefined
+    trait_count?: number | null | undefined
+    order?: number | null | undefined
+  }
+  const metadata_description = 'A Nengajo sent using HENKAKU Nengajo.'
+  const metadata_external_url = 'https://henkaku-nengajo.vercel.app'
+
+  useEffect(() => {
+    if (imageUri === '') return
+    if (metadataUri !== '') return
+    if (address === undefined) return
+    if (metadataName === '') return
+
+    const putMetadataOnIPFS = async (metadata: NengajoTokenMetadata) => {
+      console.log('Putting metadata on IPFS', metadata)
+      const url = IPFS_API_ENDPOINT + '/pinning/pinJSONToIPFS'
+
+      const pinataRequest = await fetch(url, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          pinata_api_key: `${IPFS_API_KEY}`,
+          pinata_secret_api_key: `${IPFS_API_SECRET}`
+        },
+        body: JSON.stringify(metadata)
+      })
+      var res = await pinataRequest.json()
+      setMetadataUri(`https://gateway.pinata.cloud/ipfs/${res.IpfsHash}`)
+      console.log(
+        'Metadata IPFS URI:',
+        `https://gateway.pinata.cloud/ipfs/${res.IpfsHash}`
+      )
```

Comment:
After fetching, change "setIsLoading" to false.

Reply:
I don't think we need to do this unless we change the way the page works now.

Only the submit button cares about `isLoading`, and I am hiding the button after it succeeds.

Reply:
Oh, Indeed it is.
However, We need to change "setIsLoading" to false when it fails.

Reply:
Done on https://github.com/henkaku-center/henkaku-nengajo-frontend/pull/37/commits/e3d7ef37bf4513975ea6ec8c59ebdd8243920ea6

Thank you very much for the review!

Reply:
LGTM.